packages/common/src/interfaces.ts

```diff
@@ -62,3 +62,11 @@ export type ActivityInterface = Record<string, ActivityFunction>;
  * Mapping of Activity name to function
  */
 export type UntypedActivities = Record<string, ActivityFunction>;
+
+/**
+ * A workflow history with it's ID. Useful for replay.
+ */
+export interface HistoryAndWorkflowID {
+  workflowID: string;
+  history: temporal.api.history.v1.History | unknown;
```

Comment:
```suggestion
export interface WorkflowHistory {
  workflowId: string;
  history: temporal.api.history.v1.History;
```

Suggestions:

* Change `workflowID` to `workflowId`
* Call this `WorkflowHistory` and use it as the public-facing wrapper around history everywhere. Meaning, change `fetchHistory` to return `Promise<WorkflowHistory>` and remove `FetchedHistory` interface
* This interface should never be allowed to have an unset history
* Potentially expose some kind of helpers to serialize to/from JSON (but accept workflow ID on deserialization since the JSON format likely needs to be UI/tctl compatible)

I will probably be doing similar in Python (though I will probably be putting `WorkflowHistory` in the `client` package, not a common area, but the surface area is much smaller on that project).

Reply:
The two types being separate is desirable b/c I don't want to force users to break out a `fromJson` kinda step separately when I can handle that nicely for them w/o issue.

I have passed through the id attached to `FetchedHistory` though.